trino/trino.go

```diff
@@ -1187,6 +1247,15 @@ func (st *driverStmt) exec(ctx context.Context, args []driver.NamedValue) (*stmt
 					st.user = headerValue
 				}
 
+				if arg.Name == trinoRoleHeader {
+					formattedRoles, err := parseAndFormatRoles(headerValue)
```

Comment:
We should parse it the same way as in `ParseDSN`, but I think it's actually better to require a map being passed here, and use `formatRolesFromMap()`. If we allow strings, it's not clear which format to use.